src/lib/detect/detect-breakpoint.ts

```diff
@@ -119,7 +119,13 @@ const initMediaQueryListeners = () => {
 			const listener = (mql: MediaQueryListEvent | MediaQueryList) =>
 				mql.matches && updateBreakpoint(bp);
 
-			mql.addEventListener('change', listener);
+			try {
+				mql.addEventListener('change', listener);
+			} catch {
+				//addListener is a deprecated method but Safari 13 and earlier versions do not support the new
+				//method, so we need it here as a fallback to load ads on those versions of Safari
+				mql.addListener(listener);
+			}
```

Comment:
Wondering if it is preferable to check for the existence of `addEventListener` rather than catching the exception? I think a catch would catch any exception occurring from the listener whereas a check of whether the method exists or not is slightly more explicit. Something like this (but please verify!): 

```js
if (typeof(mql.addEventListener) !== "undefined") {
  mql.addEventListener('change', listener);
} else if (typeof(mql.addListener) !== "undefined") {
  mql.addListener(listener);
}
```

Reply:
Great point! I like that way of writing it out, will give it a go